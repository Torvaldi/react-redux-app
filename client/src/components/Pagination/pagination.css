.disable_pagination {
    cursor: not-allowed;
    background-color: #eeeeee;
}

.default_pagination {
    font-size: 2rem;
}