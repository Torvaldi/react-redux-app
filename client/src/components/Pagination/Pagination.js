import React from 'react';
import { Link } from 'react-router-dom';
import './pagination.css';

import propTypes from 'prop-types';

const Pagination = ({left, right, max, current}) => {
  const parameter ="?page=";

  let cssLeft = 'hover_default';
  let cssRight = 'hover_default';

  if(current === 0 && left === 0){
    cssLeft = 'disable_pagination';
  }

  if(current === max){
    cssRight = 'disable_pagination';
  }

  return (
      <section>
        <Link to={parameter + left}>
          <i className={"material-icons default_pagination " + cssLeft}>keyboard_arrow_left</i>
        </Link>
        <Link to={parameter + right}>
          <i className={"material-icons default_pagination " + cssRight}> keyboard_arrow_right</i>
        </Link>
      </section>
    
  );
}

Pagination.propTypes = {
  left: propTypes.number.isRequired,
  right: propTypes.number.isRequired,
  max: propTypes.number.isRequired,
  current: propTypes.number.isRequired,
}

export default Pagination;
