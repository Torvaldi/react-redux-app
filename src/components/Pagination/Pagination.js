import React from 'react';
import { Link } from 'react-router-dom';
import './pagination.css';

const Pagination = ({left, right, max, current}) => {
  const parameter ="?page=";

  let cssLeft = '';
  let cssRight = '';

  if(current == 0 && left == 0){
    cssLeft = 'disable_pagination';
  }

  if(current == max && right == max){
    cssRight = 'disable_pagination';
  }

  return (
      <section>
        <Link to={parameter + left}>
          <i className={"material-icons default_pagination " + cssLeft}>keyboard_arrow_left</i>
        </Link>
        <Link to={parameter + right}>
          <i className={"material-icons default_pagination " + cssRight}> keyboard_arrow_right</i>
        </Link>
      </section>
    
  );
}



export default Pagination;
