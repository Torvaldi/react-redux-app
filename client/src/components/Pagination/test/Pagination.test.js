import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';
import Pagination from '../Pagination';

configure({ adapter: new Adapter() });

describe('<Pagination />', () => {
    
    // snapshot & props
    it('should render Pagination Componenent', () => {
        let left = 1;
        let right = 3;
        let max = 3;
        let current =  2;
        const wrapper = shallow(<Pagination left={left} right={right} max={max} current={current} />);
        expect(wrapper).toMatchSnapshot();
    });
    
    
});