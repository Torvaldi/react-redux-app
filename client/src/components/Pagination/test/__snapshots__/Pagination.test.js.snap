// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render Pagination Componenent 1`] = `ShallowWrapper {}`;
